File: leetcode3/변지협/3898. Find the Degree of Each Vertex.py


'''
1. 아이디어 :
2. 시간복잡도 :
    O(n^2)
3. 자료구조/알고리즘 :
'''

class Solution:
    def findDegrees(self, matrix: list[list[int]]) -> list[int]:
        n = len(matrix)
        ans = [0] * n
        
        for m in matrix:
            for i in range(len(m)):
                ans[i] += m[i]

        return ans

File: leetcode3/이진희/3898. Find the Degree of Each Vertex.java
/*

1. 아이디어 : 각 정점의 개수 구하기

2. 시간복잡도 : O(N^2)

3. 자료구조/알고리즘 : 완전탐색

 */

class Solution {
    public int[] findDegrees(int[][] matrix) {
        int n = matrix[0].length;

        int ans[] = new int[n];
        for(int k=0; k<n; k++) {
            for(int i=0; i<n; i++) {
                if(matrix[k][i] == 0) continue;
                ans[k]++;
            }
        }
        return ans;
    }
}